sirepo/template/srw.py

```diff
@@ -1286,6 +1255,52 @@ def _enum_text(name, model, field):
     return ''
 
 
+def _export_rsopt_config(data):
+    v = _rsopt_jinja_context(data.models.exportRsOpt)
+    filename = 'rsOptExport.zip'
+    f = re.sub(r'[^\w.]+', '-', pkio.py_path(filename).purebasename).strip('-')
+    fz = pkio.py_path(filename)
+    v.runDir = f'{f}_scan'
+    v.fileBase = f
+    tf = {k: PKDict(filename=f'{f}.{k}') for k in ['py', 'sh', 'yml']}
```

Comment:
I don't know what "tf" means. I think there are too many temporaries in this function. Better to modularize.
This for loop and the next could be combined.

Reply:
They cannot be combined - the shell script jinja needs the other filenames to be in the context. I could reorder the list but that introduces fragility. Anyway [075b377](https://github.com/radiasoft/sirepo/pull/4057/commits/075b377ec4e22e26dac2d429ee1d02791ddfeaf1)